Functional test GET /accounts fails randomly

### Expected behavior
Account endpoint should pass consistently

### Actual behavior
```
[2019-04-02T12:33:33.433Z]   1) GET /accounts
[2019-04-02T12:33:33.433Z]        ?
[2019-04-02T12:33:33.433Z]          sort
[2019-04-02T12:33:33.433Z]            using sort = balance:desc should return accounts in descending order by balance:
[2019-04-02T12:33:33.433Z] 
[2019-04-02T12:33:33.433Z]       AssertionError: expected [ Array(10) ] to deeply equal [ Array(10) ]
[2019-04-02T12:33:33.433Z]       + expected - actual
[2019-04-02T12:33:33.433Z] 
[2019-04-02T12:33:33.433Z]            "secondPublicKey": ""
[2019-04-02T12:33:33.433Z]            "unconfirmedBalance": "7990000000"
[2019-04-02T12:33:33.433Z]          }
[2019-04-02T12:33:33.433Z]          {
[2019-04-02T12:33:33.433Z]       -    "address": "17252750682695765210L"
[2019-04-02T12:33:33.433Z]       -    "balance": "600000000"
[2019-04-02T12:33:33.433Z]       -    "delegate": {
[2019-04-02T12:33:33.433Z]       -      "approval": 0
[2019-04-02T12:33:33.433Z]       -      "missedBlocks": 0
[2019-04-02T12:33:33.433Z]       -      "producedBlocks": 0
[2019-04-02T12:33:33.433Z]       -      "productivity": 0
[2019-04-02T12:33:33.433Z]       -      "rank": [null]
[2019-04-02T12:33:33.433Z]       -      "rewards": "0"
[2019-04-02T12:33:33.433Z]       -      "username": "mffljblsgj"
[2019-04-02T12:33:33.433Z]       -      "vote": "0"
[2019-04-02T12:33:33.433Z]       -    }
[2019-04-02T12:33:33.433Z]       -    "publicKey": "0ccd1957c05bcce490e605a5e56c116278e94d91f44137f5f86bb4406f2118f9"
[2019-04-02T12:33:33.433Z]       -    "secondPublicKey": ""
[2019-04-02T12:33:33.433Z]       -    "unconfirmedBalance": "600000000"
[2019-04-02T12:33:33.433Z]       -  }
[2019-04-02T12:33:33.433Z]       -  {
[2019-04-02T12:33:33.433Z]            "address": "1718609950211528553L"
[2019-04-02T12:33:33.433Z]            "balance": "3900000000"
[2019-04-02T12:33:33.433Z]            "publicKey": "6b51db4f7ba8dd2f5de4f3ff22baccb6f9a599d83a09d08217fa7d9cce3289e6"
[2019-04-02T12:33:33.433Z]            "secondPublicKey": ""
[2019-04-02T12:33:33.433Z] --
[2019-04-02T12:33:33.433Z]            "publicKey": ""
[2019-04-02T12:33:33.433Z]            "secondPublicKey": ""
[2019-04-02T12:33:33.433Z]            "unconfirmedBalance": "1500000000"
[2019-04-02T12:33:33.433Z]          }
[2019-04-02T12:33:33.433Z]       +  {
[2019-04-02T12:33:33.433Z]       +    "address": "17252750682695765210L"
[2019-04-02T12:33:33.433Z]       +    "balance": "600000000"
[2019-04-02T12:33:33.433Z]       +    "delegate": {
[2019-04-02T12:33:33.433Z]       +      "approval": 0
[2019-04-02T12:33:33.433Z]       +      "missedBlocks": 0
[2019-04-02T12:33:33.433Z]       +      "producedBlocks": 0
[2019-04-02T12:33:33.433Z]       +      "productivity": 0
[2019-04-02T12:33:33.433Z]       +      "rank": [null]
[2019-04-02T12:33:33.433Z]       +      "rewards": "0"
[2019-04-02T12:33:33.433Z]       +      "username": "mffljblsgj"
[2019-04-02T12:33:33.433Z]       +      "vote": "0"
[2019-04-02T12:33:33.433Z]       +    }
[2019-04-02T12:33:33.433Z]       +    "publicKey": "0ccd1957c05bcce490e605a5e56c116278e94d91f44137f5f86bb4406f2118f9"
[2019-04-02T12:33:33.433Z]       +    "secondPublicKey": ""
[2019-04-02T12:33:33.433Z]       +    "unconfirmedBalance": "600000000"
[2019-04-02T12:33:33.433Z]       +  }
[2019-04-02T12:33:33.433Z]        ]
[2019-04-02T12:33:33.433Z]       
[2019-04-02T12:33:33.433Z]       at accountsEndpoint.makeRequest.then.res (framework/test/mocha/functional/http/get/accounts/accounts.js:330:15)
[2019-04-02T12:33:33.433Z]       at tryCatcher (node_modules/bluebird/js/release/util.js:16:23)
[2019-04-02T12:33:33.433Z]       at Promise._settlePromiseFromHandler (node_modules/bluebird/js/release/promise.js:512:31)
[2019-04-02T12:33:33.433Z]       at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:569:18)
[2019-04-02T12:33:33.433Z]       at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
[2019-04-02T12:33:33.433Z]       at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:694:18)
[2019-04-02T12:33:33.433Z]       at _drainQueueStep (node_modules/bluebird/js/release/async.js:138:12)
[2019-04-02T12:33:33.433Z]       at _drainQueue (node_modules/bluebird/js/release/async.js:131:9)
[2019-04-02T12:33:33.433Z]       at Async._drainQueues (node_modules/bluebird/js/release/async.js:147:5)
[2019-04-02T12:33:33.433Z]       at Immediate.Async.drainQueues [as _onImmediate] (node_modules/bluebird/js/release/async.js:17:14)
```
### Steps to reproduce
Run functional get test multiple times

### Which version(s) does this affect? (Environment, OS, etc...)
1.6.0-alpha.4-

--- a/framework/test/mocha/functional/http/get/accounts/accounts.js
+++ b/framework/test/mocha/functional/http/get/accounts/accounts.js
@@ -303,14 +303,20 @@ describe('GET /accounts', () => {
 
 			it('using no sort return accounts sorted by both balance and address in asending order as default behavior', async () => {
 				return accountsEndpoint.makeRequest({}, 200).then(res => {
-					const balances = _.clone(res.body.data);
+					const balances = _.cloneDeep(res.body.data);
 					expect(
 						balances.sort((a, b) => {
 							const aBignumBalance = new Bignum(a.balance);
-							return (
-								aBignumBalance.minus(b.balance) ||
-								a.address.localeCompare(b.address)
-							);
+
+							let bignumSort = 0;
+							if (aBignumBalance.gt(b.balance)) {
+								bignumSort = 1;
+							}
+							if (aBignumBalance.lt(b.balance)) {
+								bignumSort = -1;
+							}
+
+							return bignumSort || +a.address.localeCompare(b.address);
 						})
 					).to.be.eql(res.body.data);
 				});
@@ -320,14 +326,20 @@ describe('GET /accounts', () => {
 				return accountsEndpoint
 					.makeRequest({ sort: 'balance:asc' }, 200)
 					.then(res => {
-						const balances = _.clone(res.body.data);
+						const balances = _.cloneDeep(res.body.data);
 						expect(
 							balances.sort((a, b) => {
 								const aBignumBalance = new Bignum(a.balance);
-								return (
-									aBignumBalance.minus(b.balance) ||
-									a.address.localeCompare(b.address)
-								);
+
+								let bignumSort = 0;
+								if (aBignumBalance.gt(b.balance)) {
+									bignumSort = 1;
+								}
+								if (aBignumBalance.lt(b.balance)) {
+									bignumSort = -1;
+								}
+
+								return bignumSort || +a.address.localeCompare(b.address);
 							})
 						).to.be.eql(res.body.data);
 					});
@@ -337,14 +349,20 @@ describe('GET /accounts', () => {
 				return accountsEndpoint
 					.makeRequest({ sort: 'balance:desc' }, 200)
 					.then(res => {
-						const balances = _.clone(res.body.data);
+						const balances = _.cloneDeep(res.body.data);
 						expect(
 							balances.sort((a, b) => {
-								const bBignumBalance = new Bignum(b.balance);
-								return (
-									bBignumBalance.minus(a.balance) ||
-									a.address.localeCompare(b.address)
-								);
+								const aBignumBalance = new Bignum(a.balance);
+
+								let bignumSort = 0;
+								if (aBignumBalance.gt(b.balance)) {
+									bignumSort = -1;
+								}
+								if (aBignumBalance.lt(b.balance)) {
+									bignumSort = 1;
+								}
+
+								return bignumSort || +a.address.localeCompare(b.address);
 							})
 						).to.be.eql(res.body.data);
 					});

--- a/framework/test/mocha/functional/http/get/accounts/accounts.js
+++ b/framework/test/mocha/functional/http/get/accounts/accounts.js
@@ -308,15 +308,14 @@ describe('GET /accounts', () => {
 						balances.sort((a, b) => {
 							const aBignumBalance = new Bignum(a.balance);
 
-							let bignumSort = 0;
 							if (aBignumBalance.gt(b.balance)) {
-								bignumSort = 1;
+								return 1;
 							}
 							if (aBignumBalance.lt(b.balance)) {
-								bignumSort = -1;
+								return -1;
 							}
 
-							return bignumSort || +a.address.localeCompare(b.address);
+							return a.address.localeCompare(b.address);
 						})
 					).to.be.eql(res.body.data);
 				});
@@ -331,15 +330,14 @@ describe('GET /accounts', () => {
 							balances.sort((a, b) => {
 								const aBignumBalance = new Bignum(a.balance);
 
-								let bignumSort = 0;
 								if (aBignumBalance.gt(b.balance)) {
-									bignumSort = 1;
+									return 1;
 								}
 								if (aBignumBalance.lt(b.balance)) {
-									bignumSort = -1;
+									return -1;
 								}
 
-								return bignumSort || +a.address.localeCompare(b.address);
+								return a.address.localeCompare(b.address);
 							})
 						).to.be.eql(res.body.data);
 					});
@@ -354,15 +352,14 @@ describe('GET /accounts', () => {
 							balances.sort((a, b) => {
 								const aBignumBalance = new Bignum(a.balance);
 
-								let bignumSort = 0;
 								if (aBignumBalance.gt(b.balance)) {
-									bignumSort = -1;
+									return -1;
 								}
 								if (aBignumBalance.lt(b.balance)) {
-									bignumSort = 1;
+									return 1;
 								}
 
-								return bignumSort || +a.address.localeCompare(b.address);
+								return a.address.localeCompare(b.address);
 							})
 						).to.be.eql(res.body.data);
 					});

